docs - saved question page - closes #14198 (#14274)
metabase/metabase: docs/users-guide/saved-questions.md

We could create this question using the notebook editor, like so:

![Example notebook](images/saved-questions/example-notebook.png).

Alternatively, we could create that question using SQL:

```
SELECT *
FROM   orders AS o
       INNER JOIN products AS p
               ON o.product_id = p.id
WHERE  p.category = 'Gizmo'
       AND o.created_at BETWEEN '2019-01-01' AND '2019-12-31'
```

We'll save that question as "Gizmo orders in 2019".

Now let's refer to that "Gizmo orders in 2019" in a new query. To keep it simple, let's say we just want to count all of the Gizmo orders from 2019. We can use the `#` symbol to refer to a saved question in a query.

If we type out:

```
SELECT count(*)
FROM {% raw %}{{#{% endraw %}
```

Metabase will slide out a sidebar where we can select a question to reference. We'll search for our "Gizmo orders in 2019" question

![Select a question from the variable sidebar](images/saved-questions/variable-sidebar.png)

We'll select our "Gizmo orders in 2019" question, and Metabase will update our code with the question's ID, `5`:

```
SELECT count(*)
FROM {% raw %}{{#5}}{% endraw %}
```

> This query returns the number of records from our saved question, "Gizmo orders in 2019", which has an ID of `5`.

Reviewer: I think you can cut back on the explanation a bit; seems obvious if they're following along, no?

```suggestion
This query returns the number of rows in our saved question.
```

## Saved question as a Common Table Expression (CTE)

The same syntax can be used in [Common Table Expressions (CTEs)](https://en.wikipedia.org/wiki/Hierarchical_and_recursive_queries_in_SQL#Common_table_expression) (with SQL databases that support CTEs):

```
WITH 2019_gizmo_orders AS {% raw %}{{#5}}{% endraw %}
SELECT count(*)
FROM 2019_gizmo_orders
```

> The `{% raw %}{{#5}}{% endraw %}` tag is substituted for the SQL query of the referenced question, surrounded by parentheses, equivalent to:

Reviewer: I think this could be clarified:

```suggestion
When this query is run, the `{% raw %}{{#5}}{% endraw %}` tag will be substituted with the SQL query of the referenced question, surrounded by parentheses. So it'll look like this under the hood:
```

```
WITH 2019_gizmo_orders AS (SELECT *
FROM   orders AS o
       INNER JOIN products AS p
               ON o.product_id = p.id
WHERE  p.category = 'Gizmo'
       AND o.created_at BETWEEN '2019-01-01' AND '2019-12-31')
SELECT count(*)
FROM 2019_gizmo_orders
```

# How to find a question's ID

- Selecting a question from the variable sidebar in the SQL editor will automatically add the ID number to the variable in our query.

> - We can also load the question we'd like to reference and find the ID at the end of the URL in your browser's address bar, after `/question/`, e.g., for `https://our.metabase.com/question/12345`, the question's ID would be `12345`.

Reviewer: ```suggestion
- You can also navigate to the question you'd like to reference and find its ID at the end of the URL in your browser's address bar, after `/question/`. E.g., for `https://metabase.example.com/question/12345`, the question's ID would be `12345`.
```

> ## Use cases for saved questions

Reviewer: ```suggestion
## When and why to use saved questions as a data source
```

> Why you would use a saved question as a data source:

Reviewer: ```suggestion
```

> - You can't create a view in the database, as saved questions effectively act as views. If you can create a view, and you expect that you and others will frequently query the results, consider creating a materialized view. The results will be stored in the database (as opposed to computed each time), which will speed up query time.

Reviewer: ```suggestion
- If you can't create a view in the database, since saved questions effectively act as views. If you can create a view, and you expect that you and others will frequently query the results, consider creating a materialized view. The results will be stored in the database (as opposed to computed each time), which will speed up query time.
```

> - To simplify or standardize queries for people. If you have data split across multiple tables, you can perform those complicated joins once, and provide the results as a simplified "table" that people can query. For other ways to standardize analytics, check out [segments and metrics](../administration-guide/07-segments-and-metrics.md).

Reviewer: Probably should add a link to SQL snippets docs or Learn article, too.
